File: JAVA/final-BT-java/src/services/CategorieService.java
package services;

import models.Categorie;
import repositories.CategorieRepository;
import session.Session;

import java.util.List;

public class CategorieService {
    private final CategorieRepository categorieRepository;
    public CategorieService() {
        categorieRepository = new CategorieRepository();
    }
    public boolean ajouter(Categorie categorie) {
        categorie.setIdUtilisateur(Session.getInstance().getUtilisateurConnecte().getIdUtilisateur());
        categorie.setEstCommune(false);
        return categorieRepository.ajouter(categorie) > 0;
    }
    public boolean modifier(Categorie categorie) {
        return categorieRepository.modifier(categorie) > 0;
    }
    public boolean supprimer(int idCategorie) {
        return categorieRepository.supprimer(idCategorie) > 0;
    }
    public List<Categorie> listerUtilisateur() {
        int idUtilisateur =Session.getInstance().getUtilisateurConnecte().getIdUtilisateur();
        return categorieRepository.listerUtilisateur(idUtilisateur);
    }
    public List<Categorie> listerCommunes() {
        return categorieRepository.listerCommunes();
    }
}

File: PHP/src/Services/CategorieService.php
<?php

namespace App\Services;

use App\Entity\Categorie;
use App\Repository\CategorieRepository;

class CategorieService
{
    private CategorieRepository $categorieRepository;

    public function __construct()
    {
        $this->categorieRepository = new CategorieRepository();
    }

    public function lister(int $idUtilisateur): array
    {
        return $this->categorieRepository->lister($idUtilisateur);
    }

    public function rechercherParId(int $idCategorie): ?Categorie
    {
        return $this->categorieRepository->rechercherParId($idCategorie);
    }

    public function ajouter(Categorie $categorie): bool
    {
        return $this->categorieRepository->ajouter($categorie);
    }

    public function modifier(Categorie $categorie): bool
    {
        return $this->categorieRepository->modifier($categorie);
    }

    public function supprimer(int $idCategorie): bool
    {
        return $this->categorieRepository->supprimer($idCategorie);
    }
    public function creerCategoriesParDefaut(
        int $idUtilisateur
    ): void {
        $this->categorieRepository
            ->creerCategoriesParDefaut(
                $idUtilisateur
            );
    }
}
